src/main.rs

```diff
@@ -131,9 +152,12 @@ async fn main() {
                             }
                             Ok(TakeRequestDecision::RateSucks)
```

Comment:
As previously stated, if you treat of these enum variants (and the Err case) the same way then it's a good indicator you don't need them.

Reply:
I agree, that one should not introduce code before needing/using it, but I feel if I don't use these enum variants I would have to model different errors. But I did not want to treat these scenarios as errors.
I think the enum variants are nice because they sum up the expected outputs - and they helped. me when writing tests. 

The only unnecessary complexity I see is when treating the variants in `main`.

Reply:
In this case, please use better language than `sucks`.